buildSrc/src/main/java/com/amazonaws/blox/tasks/PostProcessCloudformation.java

```diff
@@ -0,0 +1,116 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.tasks;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.io.File;
+import java.io.IOException;
+import lombok.Getter;
+import lombok.Setter;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+/**
+ * Substitute dynamic values for Serverless::Function.CodeURI and Serverless::Api.DefinitionBody
+ * into a given Cloudformation/Serverless Application Model template.
+ *
+ * <p>Defining an API gateway API using Swagger using SAM has some limitations when using a separate
+ * swagger definition file (as declared through the DefinitionUri property): - We must hard-code the
+ * region and account ID of the arn of the lambda function it should invoke in the
+ * `x-amazon-apigateway-integration/uri` property. - If we don't hardcode the Lambda function name,
+ * we must provide it as a stage variable instead.
+ *
+ * <p>Instead, SAM supports specifying the Swagger definition inline using the DefinitionBody field.
+ * This allows us to use CloudFormation functions and references to derive the name of the Lambda
+ * function to invoke, but brings its own set of limitations: - Swagger YAML definitions almost
+ * always have at least one numeric field name (in the responses object), and CloudFormation
+ * incorrectly reads this as an integer instead of a string key.
+ *
+ * <p>The solution then is to embed the Swagger definition into the SAM Template from YAML source,
+ * but to produce the combined template in JSON format to avoid the non-string key issue.
+ *
+ * <p>In the same template, cloudformation only supports paths relative to the template file or full
+ * absolute paths for the CodeUri property of a Serverless::Function. This task will substitute in
+ * the given lambdaZipFile as an absolute path into the CodeUri property.
+ */
+@Getter
+@Setter
+public class PostProcessCloudformation extends DefaultTask {
+  /** Mapper used to read the source swagger/SAM YAML templates */
+  private ObjectMapper inputMapper = new ObjectMapper(new YAMLFactory());
+  /** Mapper used to produce combined JSON template */
+  private ObjectMapper outputMapper = new ObjectMapper(new JsonFactory());
+
+  /**
+   * The name of the API GW resource in the SAM template into which the swagger template should be
+   * embedded.
+   */
+  @Input private String apiName;
```

Comment:
Would it be useful to have an option of passing these in from command line or another script instead of only accepting input from gradle? (Maybe something that can be done later)